hooks/checkout

```diff
@@ -69,22 +69,28 @@ checkout() {
   git reset --hard
   git clean -ffxdq
 
-  timeout "${GIT_REMOTE_TIMEOUT}" git fetch -v origin HEAD || exit_code=$?
   check_timeout_exit_code "${exit_code}" "${GIT_REMOTE_TIMEOUT_EXIT_CODE}"
   git config remote.origin.fetch
-  timeout "${GIT_REMOTE_TIMEOUT}" git fetch -v --prune origin "${BUILDKITE_BRANCH}" || exit_code=$?
-  check_timeout_exit_code "${exit_code}" "${GIT_REMOTE_TIMEOUT_EXIT_CODE}"
 
   if [[ -z "${BUILDKITE_COMMIT:-}" || "${BUILDKITE_COMMIT}" == "HEAD" ]]; then
+    timeout "${GIT_REMOTE_TIMEOUT}" git fetch -v --prune origin "${BUILDKITE_BRANCH}" || exit_code=$?
```

Comment:
[nitpick]: Can we standardise how we use the `timeout` tool here and the `checkout` script that we have in `canva/canva` repo ?

IIRC, there we are still using the `--preserve-status` which it is not 100% desirable but there we are not adding the `--prune` to the `fetch` operations which I think it is desirable.